emulator: tests: exclude "programs" from tests.

marete opened the pull request:
"programs" are not tests in general and can be server binaries which
wait forever and cause the entire test suite to hang and/or
timeout.

In particular add_test should only be called for binaries that are
tests, not arbitrary programs.

TESTED=Verified that "make test" no linger tries to run (and therefore
hang on) the emulator binary.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/Unoperate/google-cloud-cpp/5)
<!-- Reviewable:end -->

dopiera reviewed (commented):
Nice catch, but I think the fix should be different. Namely we should remove the relevant `add_test` from `google/cloud/bigtable/emulator/CMakeLists.txt`

> *[Reviewable](https://reviewable.io/reviews/Unoperate/google-cloud-cpp/5#-:-OO-GPmRAoeeOP5IL2VI:bf3dc8s)* status: 0 of 1 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

marete committed:
emulator: tests: exclude "programs" from tests.

"programs" are not tests in general and can be server binaries which
wait forever and cause the entire test suite to hang and/or
timeout.

In particular add_test should only be called for binaries that are
tests, not arbitrary programs.

TESTED=Verified that "make test" no linger tries to run (and therefore
hang on) the emulator binary.

marete reviewed (commented):
Yes indeed, that is a more correct and general fix. Implemented. Now all "programs" are excluded from tests.

PTAL.

> *[Reviewable](https://reviewable.io/reviews/Unoperate/google-cloud-cpp/5#-:-OO2CmfF-C3uiBGA7LV1:b7ardrw)* status: 0 of 2 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

dopiera reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/Unoperate/google-cloud-cpp/5)* status: 0 of 2 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->
